pkg/proxy/endpoints.go

```diff
@@ -300,8 +300,8 @@ func (em EndpointsMap) apply(changes *EndpointChangeTracker, staleEndpoints *[]S
 	}
 	changes.items = make(map[types.NamespacedName]*endpointsChange)
 	metrics.EndpointChangesPending.Set(0)
-	for _, lastChangeTriggerTime := range changes.lastChangeTriggerTimes {
-		*lastChangeTriggerTimes = append(*lastChangeTriggerTimes, lastChangeTriggerTime...)
+	for k, v := range changes.lastChangeTriggerTimes {
+		(*lastChangeTriggerTimes)[k] = v
```

Comment:
we should probably append, right?

Reply:
It's not needed - it iterates over a map here already and it guarantees that keys are unique.

Reply:
It works only in the current usage pattern. Technically I can imagine that we call apply multiple times for different <changes> object and want to apply it to the same <lastChangeTriggertimes>.

So while within a single call it's correct, across multiple calls it will no longer be correct. 
As an example for previous/current fields above we're explicitly calling merge/unmerge.

So here we should be calling append.